### DISCUSSION TITLE
cURL Kibana/Elasticsearch remotely

### DISCUSSION THREAD

**Discussion by brielleyoung** [Security-Onion-Solutions/securityonion#6224] [Unsupported Versions] [answered]
Hello,

I'm hoping to get some suggestions on how to cURL either Kibana or Elasticsearch directly remotely from another machine. Currently, I am working on a project where I hope to pull data from the Security Onion machine and display the information on a web interface that is hosted by another machine. Both of these machines are on the same network.

Right now, I'm thinking of trying to cURL Kibana directly to get the information I'm looking for. I can currently do this with The Hive, for example, by accessing its site directly without logging into Security Onion first. However, when I try to access Kibana directly, my cURL request gets redirected back to Security Onion's login page. This is probably intended behavior, but is there a way to either allow the remote machine to access the site directly and/or login through Security Onion first, then access Kibana? 

It's possible that I may be making this too complicated. If there's another way to access this data rather than using cURL, such as querying Elasticsearch directly, I'm open to those suggestions as well. The main goal is just to be able to query data from said machine.

**Comment by dougburks** [comment, marked as answer]
It's probably cleaner to query Elasticsearch directly. You'll probably want to be careful to only open the firewall port for the one IP address that you want to allow Elasticsearch queries from.

https://docs.securityonion.net/en/2.3/so-allow.html

**Reply by brielleyoung** [reply]
I will take a look at that, thank you!
